File: VB/WinExample.Module/BusinessObjects/DomainObject1.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinExample.Module.BusinessObjects
    <DefaultClassOptions>
    Public Class DomainObject1
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Private _Name As String
        Public Property Name() As String
            Get
                Return _Name
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", _Name, value)
            End Set
        End Property
        <Association("DomainObject1-DomainObject2s")>
        Public ReadOnly Property DomainObject2s() As XPCollection(Of DomainObject2)
            Get
                Return GetCollection(Of DomainObject2)("DomainObject2s")
            End Get
        End Property
    End Class
End Namespace

File: CS/WinExample.Module/BusinessObjects/DomainObject1.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinExample.Module.BusinessObjects {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session)
            : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        [Association("DomainObject1-DomainObject2s")]
        public XPCollection<DomainObject2> DomainObject2s {
            get {
                return GetCollection<DomainObject2>("DomainObject2s");
            }
        }
    }
}
